app/src/chat_engine.py

```diff
@@ -402,13 +402,30 @@ class ImagineLaEngine(BaseEngine):
 - Or other kinds of support questions for the Benefit Navigator tool
 then set canned_response to: "To get support with that issue, select "Need help? Contact the support team" at the top of this chatbot to open a ticket with the operations team. You can also email us at [socialbenefithelp@imaginela.org](mailto:socialbenefithelp@imaginela.org)"
 
-If the user's question is about a referral link below, set canned_response to: \
+# Precedence of canned_response:
+# 1. Coverage list of supported benefit programs
+# 2. Password reset instructions for the Benefit Navigator
+# 3. Benefit Navigator support issue instructions
+# Only if none of the above apply should you consider referral links
+
+# Referral links section
+# Important: Be strict about when to use referral links
+# Only use a referral link if the user is explicitly asking how to obtain/access/find the exact resource described by one of the referral links below.
+# By default, prefer setting needs_context=True to search the knowledge base.
```

Comment:
The default for `needs_context` is already set to `True` on line 358. Why is it also needed here?